### DISCUSSION TITLE
handle controlled errors

### DISCUSSION THREAD

**Discussion by bsdgithub** [flowsforapex/apex-flowsforapex#722] [General]
hi,
I have a situation when I'm changing the process variable type (it's stored a response and sometimes it's bigger then 4K so should be a clob). I was trying to use controlled delete:
` begin
       flow_process_vars.delete_var(pi_prcs_id => pi_prcs_id, pi_var_name => pi_var_name, pi_scope => pi_scope);
  exception when others then
            null;
  end;`
That somehow mark an error and the step it's marked with error, even if the error doesn't apear.
What it's the best practice on that? How should I handle it?
Thanks,

**Comment by rallen2010** [comment]
Sorry - I'm not quite clear what the problem is from your description.

Are you trying to change one process variable from a varchar2 to a clob by creating a new proc var as a clob and then deleting the old (VC2) one?  But the delete_var gives you an error?

Does the new one have the same name as the old one?
Are you deleting the old one before you create the new one?  (If they have the same name you will need to do that as (prcs_id, proc_name, scope ) needs to be unique.

**Reply by bsdgithub** [reply]
Hi,
yes, practically it's the same var (same name) but, I was trying to delete and recreate from varchar2 to clob, if the response get bigger than 4k. If delete gave an error, the process is getting an error, even if it's cached.
What I did, check if the var exists, then delete it if the length it's asking for changing data type, otherwise update it.
What was strange is that even the exception block existed, delete gave an error (var doesn't exist), the process was marked with an error. My expectation was to not have any problem because the delete error was controlled.
Thanks,
bsd
